src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -139,11 +125,24 @@ public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(
      * @return a textSubmission without any manual result or an empty Optional if no submission without manual result could be found
      */
     public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(TextExercise textExercise, boolean skipAssessmentQueue, boolean examMode, int correctionRound) {
+        var assessableSubmissions = getAssessableSubmissions(textExercise, examMode, correctionRound);
+
         // If automatic assessment is enabled and available, try to learn the most possible amount during the first correction round
-        if (textExercise.isAutomaticAssessmentEnabled() && textAssessmentQueueService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
-            return textAssessmentQueueService.get().getProposedTextSubmission(textExercise);
+        if (textExercise.isFeedbackSuggestionsEnabled() && athenaSubmissionSelectionService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
+            var athenaSubmission = athenaSubmissionSelectionService.get().getProposedSubmission(textExercise, assessableSubmissions.stream().map(Submission::getId).toList());
+            if (athenaSubmission.isPresent()) {
+                // test again if it is still assessable (Athena might have taken some time to respond and another assessment might have started in the meantime)
+                var submission = textSubmissionRepository.findWithEagerResultsAndFeedbackAndTextBlocksById(athenaSubmission.get().getId());
+                if (submission.isPresent() && submission.get().getLatestResult() == null) { // submission assessable?
```

Comment:
Is this correct? I'd assume that Athena creates an automatic result that contains the feedback suggestions. 
```suggestion
                if (submission.isPresent() && (submission.get().getLatestResult() == null || !submission..get().getLatestResult().isManual())) { // submission assessable?
```

Reply:
I've changed that for more robustness 👍 I thought about changing the `!...isManual()` check to `...isAutomatic()`, but I think I also want to ignore existing semi-automatic results, so this should be correct.